### ISSUE TITLE
Library not working, infinite loop

### ISSUE THREAD

**Issue by G4mingEdition** [alexandrtovmach/react-microsoft-login#84]
Hi,

I have to create a react microsoft login so I used your library but I encoutered a bug.

**My code:**
I'm using React class components with Redux.
![image](https://user-images.githubusercontent.com/38436883/122717930-926f4a00-d26c-11eb-855c-ea48238ed721.png)
![image](https://user-images.githubusercontent.com/38436883/122717964-9c914880-d26c-11eb-93dc-0bb8427651f3.png)

**Actual result:**
When loading the page, the library is automatically showing the Microsoft's popup and looping "Trying to sign you in"
![image](https://user-images.githubusercontent.com/38436883/122718219-f6920e00-d26c-11eb-800a-18443b3863e1.png)
Then I got this error
![image](https://user-images.githubusercontent.com/38436883/122718459-42dd4e00-d26d-11eb-9ab6-ebd1089cbf9a.png)

**Additional information:**
If I use "withUserData", it's working but it's spamming my API and when I disconnect it's still spamming my API and connecting me again to the website. I'm using "react-facebook-login" and "react-google-login" and I haven't encountered this problem.

Any solution ? 😕

**Comment by alexandrtovmach** [comment]
I think you need to be more accurate when say "Library not working", because it seems more like bug on your side =)
Anyway, I see core issue in this "**When loading the page**, the library is automatically showing the Microsoft's popup and looping". Auth process should be initiated by pressing button, not by page load, so please check why it's happen or provide reproducible example.

Good info to add to issue is:
- package version that you're using
- is it reproducing on [demo website](https://alexandrtovmach.github.io/react-microsoft-login/)

**Comment by G4mingEdition** [comment]
I have fixed the issue on my project. I had to clear the session that contains all information about Microsoft.
![image](https://user-images.githubusercontent.com/38436883/122738692-81313800-d282-11eb-9df2-8a95a9a63975.png)
Without that, the library is spamming Microsoft (and my API) and call Microsoft's auth on page load.

**Comment by SamuelPinho** [comment]
@G4mingEdition thanks for the help man, it solved the same problem here.

@alexandrtovmach any ideia if this could be added by default on the lib?

**Comment by alexandrtovmach** [comment]
You not answered to my questions about package version, so I'm not sure about it

**Comment by G4mingEdition** [comment]
I've added the package version in "Additional information" (1.15.0)

**Comment by muyiwaoyeniyi** [comment]
@alexandrtovmach I'm also having the same issue after upgrading from 1.12.0 to 1.15.0. If I sign out within my app and revisit the sign in page, microsoft authentication immediately happens without clicking on the button. You have to clear the session store or close the tab and use another tab.

**Comment by rafenasir** [comment]
@G4mingEdition .Can I stop the call to authHandler function on page reload?

**Comment by muyiwaoyeniyi** [comment]
@rafenasir just do this after successful authentication

sessionStorage.removeItem("msal.idtoken")

**Comment by rafenasir** [comment]
Thanks a lot!! :)

**Comment by alexandrtovmach** [comment]
looks like a hacky solution)
will check if there a proper way to handle that

**Comment by rafenasir** [comment]
> looks like a hacky solution) will check if there a proper way to handle that

@alexandrtovmach yes!! In the above scenario if I remove the msal.idToken it doesn't show the logout button :) I would highly appreciate if you can come up with a proper solution.

**Comment by muyiwaoyeniyi** [comment]
@alexandrtovmach I pasted the source of the issue here https://github.com/alexandrtovmach/react-microsoft-login/issues/84#issuecomment-895918353 ... it would be helpful to allow us send a prop to enable or disable this behavior.

**Comment by ishwark7** [comment]
Hi @alexandrtovmach ,
Facing this issue in laster 1.15.0. 
Using sessionStorage.removeItem("msal.idtoken") as of now to avoid the loop. Please let us know if this can be handled in a better way. Thanks
